core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -259,6 +259,7 @@ class KafkaServer(
 
         /* generate brokerId */
         config.brokerId = getOrGenerateBrokerId(initialMetaPropsEnsemble)
+        config.nodeId = config.brokerId
```

Comment:
Could you please add comments to explain why zk broker needs to sync node.id here

Reply:
Sure.